main.cpp

```diff
@@ -43,27 +43,36 @@ int main() {
     // Open a socket on the network interface, and create a TCP connection to ifconfig.io
     TCPSocket socket;
     // Send a simple http request
+    char hostname[] = "ifconfig.io";
     char sbuffer[] = "GET / HTTP/1.1\r\nHost: ifconfig.io\r\nConnection: close\r\n\r\n";
     nsapi_size_t size = strlen(sbuffer);
 
     result = socket.open(net);
     if (result != 0) {
         printf("Error! socket.open() returned: %d\n", result);
+        goto DISCONNECT;
     }
 
-    result = net->gethostbyname("ifconfig.io", &a);
+    // Get the host address
+    printf("\nResolve hostname %s\n", hostname);
+    result = net->gethostbyname(hostname, &a);
     if (result != 0) {
-        printf("Error! Could not resolve host ifconfig.io. Result: %d\n", result);
+        printf("Error! gethostbyname(%s) returned: %d\n", hostname, result);
+        net->disconnect();
+        exit(-1);
     }
+    printf("%s address is %s\n", hostname, (a.get_ip_address() ? a.get_ip_address() : "None") );
+
     a.set_port(80);
     result = socket.connect(a);
     if (result != 0) {
         printf("Error! socket.connect() returned: %d\n", result);
-        goto DISCONNECT;
+        net->disconnect();
```

Comment:
Can you call socket.close() if the socket.connect() has failed? @SeppoTakalo  @anttiylitokola

Reply:
You can, but the socket's destructor will call it anyway.

Reply:
`close()` is an opposite of `open()`, but also serves as disconnect call.
So even if `connect()` fails, you should still clean up the resources allocated by `open()`